### PR TITLE
Return of Autoreplies

### PR DESCRIPTION
**Pull request by realforest2001**
# About the pull request
Re-adds AutoReplies to ahelps.

# Explain why it's good for the game
Makes things a bit easier for Mod and Admins.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
add: Re-Adds autoreplies to ahelps.
/:cl:

### CODE REVIEW

File: code/modules/admin/verbs/adminhelp.dm [hunk 1]
```diff
@@ -507,23 +507,39 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
...
+/datum/admin_help/proc/GetReplies()
+	var/list/options = list()
+	for(var/option in subtypesof(/datum/autoreply/admin))
+		var/datum/autoreply/admin/n_option = new option
+		options += n_option.title
+	return options
+
+/// Resolve ticket with a premade message
+/datum/admin_help/proc/AutoReply(key_name = key_name_admin(usr))
 	if(state != AHELP_ACTIVE)
```
Comment (harryob) [inline code/modules/admin/verbs/adminhelp.dm:520]:
```suggestion
/datum/admin_help/proc/AutoReply()
	var/key_name = key_name_admin(usr)
	if(state != AHELP_ACTIVE)
```

File: code/modules/admin/verbs/adminhelp.dm [hunk 2]
```diff
@@ -507,23 +507,39 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
 	log_ahelp(id, "Rejected", "Rejected by [usr.key]", null, usr.ckey)
 	Close(silent = TRUE)
 
-//Resolve ticket with IC Issue message
-/datum/admin_help/proc/ICIssue(key_name = key_name_admin(usr))
+/// I feel like this is a really roundabout way to do what I want, so I'm open to suggestions on making it more efficient.
```
Comment (harryob) [inline code/modules/admin/verbs/adminhelp.dm:510]: what's wrong with just using the GLOB.adminreplies list if they're already indexed by titles

File: code/modules/admin/autoreply.dm [hunk 3]
```diff
@@ -0,0 +1,36 @@
+GLOBAL_REFERENCE_LIST_INDEXED(adminreplies, /datum/autoreply/admin, title)
+
+/datum/autoreply
+	var/title = "Blank"
+	var/message = "Lorem ipsum dolor sit amit."
+	var/closer = TRUE
```
Comment (harryob) [inline code/modules/admin/autoreply.dm:6]: /// doc these

**Review by harryob — CHANGES_REQUESTED** [review]
i love datumisation so this is good

File: code/modules/admin/verbs/adminhelp.dm (hunk 2 above)
Comment (realforest2001) [reply code/modules/admin/verbs/adminhelp.dm:510]: Can't parse GLOB.adminreplies directly into a list as it can't handle the fact that it's X = Y.

Comment (harryob) [reply code/modules/admin/verbs/adminhelp.dm:510]: so

```
var/input = tgui_input_list(yada yada yada, GLOB.admin_replies
if(!input)
 return
input = GLOB.admin_replies[input]
```
doesn't work?

Comment (realforest2001) [reply code/modules/admin/verbs/adminhelp.dm]: discussed in discord

File: code/modules/admin/autoreply.dm [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	title = "Bug Report"
+	message = "Please report all bugs on our <a href='"+URL_ISSUE_TRACKER+"'>Github</a>. Administrative staff are unable to fix most bugs on a round to round basis and only round critical bugs, or exploits, should be ahelped."
+
+/datum/autoreply/admin/marine
+	title = "Marine Guide"
+	message = "Your action can be answered by the <a href='"+URL_WIKI_MARINE_QUICKSTART+"'>Marine Quickstart Guide</a>. If anything is unclear or you have another question please make a new mentorhelp or ahelp about it."
+
+/datum/autoreply/admin/xeno
+	title = "Xeno Guide"
+	message = "Your action can be answered by the <a href='"+URL_WIKI_XENO_QUICKSTART+"'>Marine Quickstart Guide</a>. If anything is unclear or you have another question please make a new mentorhelp or ahelp about it."
```
Comment (harryob) [inline code/modules/admin/autoreply.dm:31]: change name of link to xeno

File: code/modules/admin/autoreply.dm [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+/datum/autoreply/admin/changelog
+	title = "Changelog"
+	message = "The answer to your question can be found in the <a href='"+URL_CHANGELOG+"'>Changelog</a>. Click the changelog button at the top-right of the screen to view it in-game, alternatively go the the CM-SS13 discord server where you can look at the cm-changelog channel to find links to any merged changes to the server."
```
Comment (harryob) [inline code/modules/admin/autoreply.dm:35]: this doesn’t work don’t link to it

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]

**Pull request merged**